File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/composite/solution/SetupClass.java
// SetupClass.java

public class SetupClass
{
	public static Journey setup()
	{
		MultipleJourney todaysItinerary = new MultipleJourney("Today's itinerary");
	
			MultipleJourney work = new MultipleJourney("Work");
			work.addJourney(SingleJourneyFlyweightFactory.getInstance().getFlyweightFor("Apple Computer"));
	
		todaysItinerary.addJourney(work);
	
			MultipleJourney foodShopping = new MultipleJourney("Food shopping");
			foodShopping.addJourney(SingleJourneyFlyweightFactory.getInstance().getFlyweightFor("Whole Foods"));
			foodShopping.addJourney(SingleJourneyFlyweightFactory.getInstance().getFlyweightFor("Stop & Shop"));
	
		todaysItinerary.addJourney(foodShopping);
	
			MultipleJourney petShopping = new MultipleJourney("Pet shopping");
			petShopping.addJourney(SingleJourneyFlyweightFactory.getInstance().getFlyweightFor("Petco"));
			petShopping.addJourney(SingleJourneyFlyweightFactory.getInstance().getFlyweightFor("Stop & Shop"));
	
		todaysItinerary.addJourney(petShopping);
		
		return todaysItinerary;
	}
}

File: dp_labs_and_solutions/dp_labs_and_solutions/dp_cpp_labs_and_solutions/singleton/flyweight/lab/setup_class.cpp
// setup_class.cpp

#include <string>
#include "journey.h"
#include "multiple_journey.h"
#include "setup_class.h"
#include "single_journey.h"
#include "single_journey_flyweight_factory.h"

journey * setup_class::setup()
{
	multiple_journey * todays_itinerary = new multiple_journey("Today's itinerary");

		multiple_journey * work = new multiple_journey("Work");
		work->add_journey(single_journey_flyweight_factory::get_instance().get_flyweight_for("Apple Computer"));

	todays_itinerary->add_journey(work);

		multiple_journey * food_shopping = new multiple_journey("Food shopping");
		food_shopping->add_journey(single_journey_flyweight_factory::get_instance().get_flyweight_for("Whole Foods"));
		food_shopping->add_journey(single_journey_flyweight_factory::get_instance().get_flyweight_for("Stop & Shop"));

	todays_itinerary->add_journey(food_shopping);

		multiple_journey * pet_shopping = new multiple_journey("Pet shopping");
		pet_shopping->add_journey(single_journey_flyweight_factory::get_instance().get_flyweight_for("Petco"));
		pet_shopping->add_journey(single_journey_flyweight_factory::get_instance().get_flyweight_for("Stop & Shop"));

	todays_itinerary->add_journey(pet_shopping);
	
	return todays_itinerary;
}
